src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -274,11 +274,17 @@ module.exports = function<T, P, I, TI, C, CX>(
     if (updateQueue) {
       const prevState = workInProgress.memoizedState;
       const state = beginUpdateQueue(workInProgress, updateQueue, null, prevState, null, priorityLevel);
+      if (prevState === state) {
+        // If the state is the same as before, that's a bailout because we had
+        // no work matching this priority.
+        return bailoutOnAlreadyFinishedWork(current, workInProgress);
+      }
       const element = state.element;
       reconcileChildren(current, workInProgress, element);
       workInProgress.memoizedState = state;
```

Comment:
Shouldn't we return `workInProgress.child` inside this branch? I get that `bailoutOnAlreadyFinishedWork` does that anyway but if there's state then it's not really a bailout, is it?